cli: check existing instance in RocksDB debug

tbg opened the pull request:
Running, for example, `./cockroach debug keys no-such-dir` would result
in a new RocksDB instance to be initialized at `no-such-dir`. This is
confusing and annyoing; instead, return an error.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20479)
<!-- Reviewable:end -->

nvanbenschoten commented:
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[pkg/server/config.go, line 466 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479#-L-bYRrshean_0CRerrT:-L-bYRrshean_0CRerrU:bryevrc) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8e5a4bdd9c252a83fe478fd69ddab17c7a562c6/pkg/server/config.go#L466)):*
> ```Go
> 		cfg.TestingKnobs.Store.(*storage.StoreTestingKnobs).SkipMinSizeCheck
> 	for i, spec := range cfg.Stores.Specs {
> 		log.Eventf(ctx, "initializing %+v")
> ```

This seems wrong. `log.Eventf(ctx, "initializing %+v", spec)`?

---

*[pkg/storage/engine/rocksdb.go, line 301 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479#-L-bYZSSJ0OSG7QcO_i0:-L-bYZSSJ0OSG7QcO_i1:b-rwclhs) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8e5a4bdd9c252a83fe478fd69ddab17c7a562c6/pkg/storage/engine/rocksdb.go#L301)):*
> ```Go
> 	// an initialized RocksDB instance.
> 	//
> 	// Makes no sense for in-memory instances.
> ```

What happens in this case? Is it ignored or is an error thrown?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/server/config.go, line 466 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479#-L-bYRrshean_0CRerrT:-L-cunjYm4LddL3L3hTo:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8e5a4bdd9c252a83fe478fd69ddab17c7a562c6/pkg/server/config.go#L466)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

This seems wrong. `log.Eventf(ctx, "initializing %+v", spec)`?
</blockquote></details>

Done.

---

*[pkg/storage/engine/rocksdb.go, line 301 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479#-L-bYZSSJ0OSG7QcO_i0:-L-cxCmm2z-aR5DZR8J8:b4kp535) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8e5a4bdd9c252a83fe478fd69ddab17c7a562c6/pkg/storage/engine/rocksdb.go#L301)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

What happens in this case? Is it ignored or is an error thrown?
</blockquote></details>

Throws an error (added a test).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479)*
<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20479#-:-L-cxd2jdR4shFXTFBN2:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
cli: check existing instance in RocksDB debug

Running, for example, `./cockroach debug keys no-such-dir` would result
in a new RocksDB instance to be initialized at `no-such-dir`. This is
confusing and annyoing; instead, return an error.

Release note: None
